pkg/azurefile/azurefile.go

```diff
@@ -407,10 +410,29 @@ func (d *Driver) Run(endpoint, kubeconfig string, testBool bool) {
 	}
 	d.AddNodeServiceCapabilities(nodeCap)
 
-	s := csicommon.NewNonBlockingGRPCServer()
-	// Driver d act as IdentityServer, ControllerServer and NodeServer
-	s.Start(endpoint, d, d, d, testBool)
-	s.Wait()
+	//setup grpc server
+	opts := []grpc.ServerOption{
+		grpc.UnaryInterceptor(csicommon.LogGRPC),
+	}
+	server := grpc.NewServer(opts...)
+	csi.RegisterIdentityServer(server, d)
+	csi.RegisterControllerServer(server, d)
+	csi.RegisterNodeServer(server, d)
+	d.server = server
+
+	listener, err := csicommon.ListenEndpoint(endpoint)
+	if err != nil {
+		klog.Fatalf("failed to listen endpoint: %v", err)
+	}
+	go func() {
+		<-ctx.Done()
+		d.server.GracefulStop()
+	}()
+	if err = d.server.Serve(listener); errors.Is(err, grpc.ErrServerStopped) {
+		klog.Infof("gRPC server stopped serving")
```

Comment:
why not return error here? for ut pass?

Reply:
If the context expired/closed, the grpc server will return ErrServerStopped, I don't think it is an error. Because it is an expected behavior

Reply:
ok, can you add a comment for this? and pls check whether it's possible to wrap logic from L413-L434 (similar to existing code), that would make logic more clear and make it possible to write ut for such code change.

Reply:
wait, this `Run` function should never stop,  and also don't need to return error since this func never stops

Reply:
yes. it will block until the server stops.

Reply:
I believe this is covered in pkg/azurefile/azurefile_test.go https://github.com/kubernetes-sigs/azurefile-csi-driver/blob/bd8e20bf5a8721088ed8fce0fceb17ab5d243617/pkg/azurefile/azurefile_test.go#L1019-L1106